src/test/java/core/basesyntax/DataParserTest.java

```diff
@@ -0,0 +1,81 @@
+package core.basesyntax;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataParser;
+import core.basesyntax.service.impl.DataParserImpl;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class DataParserTest {
+    private DataParser dataParser;
+
+    @BeforeEach
+    public void setUp() {
+        dataParser = new DataParserImpl();
+    }
+
+    @Test
+    public void parseData_ValidData_Ok() {
+        List<String> validData = Arrays.asList(
+                "type,fruit,quantity",
+                "b,banana,20",
+                "s,apple,50",
+                "p,orange,30"
+        );
+        List<FruitTransaction> expectedTransactions = Arrays.asList(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 20),
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, "apple", 50),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "orange", 30)
+        );
+
+        List<FruitTransaction> actualTransactions = dataParser.parseData(validData);
+        assertFruitTransactionsEqual(expectedTransactions, actualTransactions);
+    }
+
+    @Test
+    public void parseData_InvalidRowFormat_NotOk() {

```

Comment:
public is redundant for test methods in the JUnit5. Fix it everywhere 
```suggestion
    void parseData_InvalidRowFormat_NotOk() {
```